examples/cai_numba.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2008-2021 Andreas Kloeckner
+# Copyright 2021 NVIDIA Corporation
+
+from numba import cuda
+
+import pycuda.driver as pycuda
+import pycuda.autoinit  # noqa
+import pycuda.gpuarray as gpuarray
+
+import numpy
+
+
+# Create a PyCUDA gpuarray
+a_gpu = gpuarray.to_gpu(numpy.random.randn(4, 4).astype(numpy.float32))
+print("original array:")
+print(a_gpu)
+
+# Retain PyCUDA context as primary and make current so that Numba is happy
+pyc_dev = pycuda.autoinit.device
+pyc_ctx = pyc_dev.retain_primary_context()
+pyc_ctx.push()
```

Comment:
This is kind of clunky IMO. (Mostly I don't quite agree with making a context in `autoinit` and then not using it, really.) Want to make a version of `autoinit` that just retains the device primary context?

Reply:
Thanks for the suggestion - I think it's a great idea, and would help simplify interoperability with other libraries like Numba, cuDF, etc., where users are often caught out by using `autoinit` then finding that Numba complains because the current context is non-primary. Will look into implementing it in this PR.

Reply:
I've added `pycuda.autoprimaryctx` and used it for the Numba example - I've tried to keep `autoprimaryctx` similar to `autoinit`, except for the retention of the context instead of the creation of a new one.